conan/api/subapi/config.py

```diff
@@ -32,6 +38,50 @@ def install(self, path_or_url, verify_ssl, config_type=None, args=None,
         configuration_install(app, path_or_url, verify_ssl, config_type=config_type, args=args,
                               source_folder=source_folder, target_folder=target_folder)
 
+    def install_pkg(self, ref, lockfile):
+        conan_api = self.conan_api
+        remotes = conan_api.remotes.list()  # ready to use remotes arguments
+        # Ready to use profiles as inputs
+        profile_host = profile_build = conan_api.profiles.get_profile([])
+
+        app = ConanApp(self.conan_api)
+        conanfile = app.loader.load_virtual(requires=[RecipeReference.loads(ref)])
+        consumer_definer(conanfile, profile_build, profile_host)
+        root_node = Node(ref=None, conanfile=conanfile, context=CONTEXT_HOST, recipe=RECIPE_VIRTUAL)
+        remotes = remotes or []
+        update = ["*"]
+        builder = DepsGraphBuilder(app.proxy, app.loader, app.range_resolver, app.cache, remotes,
+                                   update, update, self.conan_api.config.global_conf)
+        deps_graph = builder.load_graph(root_node, profile_host, profile_build, lockfile)
+        deps_graph.report_graph_error()
+        pkg = deps_graph.root.dependencies[0].dst
+        ConanOutput().info(f"Configuration from package: {pkg}")
+        if pkg.conanfile.package_type is not PackageType.CONF:
+            raise ConanException(f'{pkg.conanfile} is not of package_type="configuration"')
+        if pkg.dependencies:
+            raise ConanException(f"Configuration package {pkg.ref} cannot have dependencies")
+
+        conan_api.graph.analyze_binaries(deps_graph, None, remotes, update=update, lockfile=lockfile)
+        conan_api.install.install_binaries(deps_graph=deps_graph, remotes=remotes)
+
+        # We check if this specific version is already installed
+        config_pref = pkg.pref.repr_notime()
+        config_version_file = HomePaths(conan_api.home_folder).config_version_path
+        if os.path.exists(config_version_file):
+            version = load(config_version_file)
+            if version == config_pref:
+                ConanOutput().info(f"Package '{pkg}' already configured, "
+                                   "skipping configuration install")
+                return  # Already installed, we can skip repeating the install
+
+        uri = pkg.conanfile.package_folder
+        config_type = "dir"
+        from conans.client.conf.config_installer import configuration_install
+        app = ConanApp(self.conan_api)
```

Comment:
Why are we creating another `ConanApp` instance?

Reply:
good point, removing it